duffel_api/api/booking/offer_requests.py

```diff
@@ -85,6 +89,12 @@ def _validate_slices(slices):
             ):
                 raise OfferRequestCreate.InvalidSlice(travel_slice)
 
+    @staticmethod
+    def _validate_max_connections(max_connections):
+        """Validate the max connection number"""
+        if max_connections < 0:
```

Comment:
we only support integers in the API, I wonder if we should:

```suggestion
        if isinstance(max_connections, int) and max_connections < 0:
```

Reply:
Good shout! Assuming `isinstance` returns True in this case when it is an int, I think we would want:
`if not isinstance(max_connections, int) or max_connections < 0:`
as we raise and error when True. I'll update now

Reply:
👋 Should these kinds of validations occur in the client, e.g. should these validations occur here _and_ in the API?

Reply:
I'd agree with @sgerrand that we've generally avoided any sort of explicit type checking in the client and only performed basic consistency checks (i.e. that there are non-zero elements or certain attributes exist).